### PR TITLE
[ch29016] Added custom json Validator based on jsonschema lib

### PR DESCRIPTION
**Pull request by emaciupe**
[ch29016]

### CODE REVIEW

**Comment by emaciupe** [comment]
@domdinicola i didn't mark it as ready, please take a look and give an earlier input from your side on the approach. Thanks!

**Comment by domdinicola** [comment]
@emaciupe looks good to me! what else is missing here?

**Comment by emaciupe** [comment]
> @emaciupe looks good to me! what else is missing here?

@domdinicola test changes were committed, so i'd say nothing left out from this PR now. There's another separate card to clean up the indicator data.

File: django_api/etools_prp/apps/core/validators.py [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 from rest_framework.exceptions import ValidationError
 
-from etools_prp.apps.cluster.models import ClusterActivity, ClusterObjective
-from etools_prp.apps.partner.models import PartnerActivity, PartnerActivityProjectContext, PartnerProject
-
 
 class AddIndicatorObjectTypeValidator:
 
     def __call__(self, value):
+        from etools_prp.apps.cluster.models import ClusterActivity, ClusterObjective
```
[the remark this answers is not in the document]
Comment (domdinicola) [reply django_api/etools_prp/apps/core/validators.py:13]: weird.... code changes should not have impacted that...

**Comment by domdinicola** [comment]
yep... we should write a script to clean the values and/or flag the results that need to be fixed manually

**Pull request merged**